backend/routes/generations.py

```
@@ -78,15 +78,29 @@ async def generate_speech(
+        # so the DB row records the actual spoken text before enqueueing.
+        from ..backends import get_llm_backend
+
+        llm_backend = get_llm_backend()
+        if getattr(llm_backend, "supports_streaming", lambda: False)():
+            personality_prompt = profile.personality
+            source = "personality_speak"
+        else:
+            try:
+                llm_result = await personality.rewrite_as_profile(profile.personality, data.text)
+            except ValueError as e:
+                raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)